packages/release-please/src/release-please.ts

```diff
@@ -383,13 +385,42 @@ async function runBranchConfiguration(
       plugins
     );
     try {
-      const numReleases = await Runner.createReleases(manifest);
-      logger.info(`Created ${numReleases} releases`);
-      if (numReleases > 0) {
+      const releases = await Runner.createReleases(manifest);
+      logger.info(`Created ${releases.length} releases`);
+      if (releases.length > 0) {
         // we created a release, reload config which may include the latest
         // version
         manifest = null;
       }
+      if (branchConfiguration.tagPullRequestNumber) {
+        // Record the pull request number to the commit as
+        // a tag.
+
+        // It's possible for manifest.createReleases() to create
+        // releases for multiple pull requests. Find the unique
+        // pair of pull request numbers and their commit SHAs.
+        const prNumberToSha = new Map<number, string>();
+        for (const release of releases) {
+          prNumberToSha.set(release.prNumber, release.sha);
+        }
+        for (const prNumber of prNumberToSha.keys()) {
```

Comment:
For iterating over a map, you can do something like: 
`for (let [key, value] of  map.entries()) {for (let [key, value] of  map.entries()) {`

Reply:
Updated. Thanks.